src/query.c

```diff
@@ -1093,16 +1097,13 @@ static void tag_strtolower(char **pstr, size_t *len, int caseSensitive) {
   *str = '\0';
 
   if (!caseSensitive) {
-    // TODO: MOD-8799 Support longer dst
-    size_t newLen = 0;
-    char *dst = unicode_tolower(origStr, length, &newLen);
+    char *dst = unicode_tolower(origStr, &length);
```

Comment:
We can consider adding a `capacity` argument, so if unescaped and lowered we might still not need to reallocate due to the extra space of the escape characters that could be used (`length` is actually smaller than the capacity)